internal/operands/vm-console-proxy/reconcile.go

```diff
@@ -0,0 +1,238 @@
+package vm_console_proxy
+
+import (
+	"fmt"
+	"strconv"
+
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"kubevirt.io/ssp-operator/internal/common"
+	"kubevirt.io/ssp-operator/internal/operands"
+	vm_console_proxy_bundle "kubevirt.io/ssp-operator/internal/vm-console-proxy-bundle"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	operandName      = "vm-console-proxy"
+	operandComponent = "vm-console-proxy"
+	enableAnnotation = "ssp.kubevirt.io/enable-vm-console-proxy"
+)
+
+// Define RBAC rules needed by this operand:
+// +kubebuilder:rbac:groups=core,resources=services;serviceaccounts;configmaps,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles;clusterrolebindings,verbs=get;list;watch;create;update;patch;delete
+
+// RBAC for created roles
+// +kubebuilder:rbac:groups=kubevirt.io,resources=virtualmachineinstances,verbs=get;list;watch
+// +kubebuilder:rbac:groups=authentication.k8s.io,resources=tokenreviews,verbs=create
+// +kubebuilder:rbac:groups=authorization.k8s.io,resources=subjectaccessreviews,verbs=create
+
+type vmConsoleProxy struct {
+	serviceAccount     *core.ServiceAccount
+	clusterRole        *rbac.ClusterRole
+	clusterRoleBinding *rbac.ClusterRoleBinding
+	service            *core.Service
+	deployment         *apps.Deployment
+	configMap          *core.ConfigMap
+}
+
+var _ operands.Operand = &vmConsoleProxy{}
+
+func New(bundle *vm_console_proxy_bundle.Bundle) *vmConsoleProxy {
+	return &vmConsoleProxy{
+		serviceAccount:     &bundle.ServiceAccount,
+		clusterRole:        &bundle.ClusterRole,
+		clusterRoleBinding: &bundle.ClusterRoleBinding,
+		service:            &bundle.Service,
+		deployment:         &bundle.Deployment,
+		configMap:          &bundle.ConfigMap,
+	}
+}
+
+func (v *vmConsoleProxy) Name() string {
+	return operandName
+}
+
+func (v *vmConsoleProxy) WatchTypes() []operands.WatchType {
+	return []operands.WatchType{
+		{Object: &core.ServiceAccount{}},
+		{Object: &core.Service{}},
+		{Object: &apps.Deployment{}},
+		{Object: &core.ConfigMap{}},
+	}
+}
+
+func (v *vmConsoleProxy) WatchClusterTypes() []operands.WatchType {
+	return []operands.WatchType{
+		{Object: &rbac.ClusterRole{}},
+		{Object: &rbac.ClusterRoleBinding{}},
+	}
+}
+
+func (v *vmConsoleProxy) RequiredCrds() []string {
+	return nil
+}
+
+func (v *vmConsoleProxy) Reconcile(request *common.Request) ([]common.ReconcileResult, error) {
+	var results []common.ReconcileResult
+	var reconcileFunc []common.ReconcileFunc
+
+	if !isEnabled(request) {
+		v.Cleanup(request)
```

Comment:
The return value needs to be handled. At least the error.

Reply:
Yep, sure thing. Thanks!